### PR TITLE
Turned deparseAndTruncate() into deparseAndFlatten() with a sensible …

### PR DESCRIPTION
**Pull request by nealrichardson**
…default for truncation (40 char), if set to NULL no truncation will occur.

### CODE REVIEW

**Comment by nealrichardson** [comment]
Travis failed because I was tweaking 'httptest' over the weekend and needed to pin an old version. Merge master into your branch and travis should pass.

File: R/show.R [hunk 1]
```diff
@@ -191,10 +191,15 @@ expressionValue <- function (expr) {
...
+deparseAndFlatten <- function (x, max_length = 40, ...) {
     out <- deparse(x, ...)
     if (length(out) > 1) {
-        out <- paste0(out[1], "...")
+        out <- paste0(out, collapse="")
+    }
+    # if max_length is null, do nothing
+    # else return 1:max_length of out
+    if (!is.null(max_length)) {
+        out <- substr(out,1,max_length)
```
Comment (nealrichardson) [inline R/show.R]: space after comma please

File: R/show.R [hunk 2]
```diff
@@ -191,10 +191,15 @@ expressionValue <- function (expr) {
     unlist(expr$column %||% expr$value)
 }
 
-deparseAndTruncate <- function (x, ...) {
+deparseAndFlatten <- function (x, max_length = 40, ...) {
```
Comment (nealrichardson) [inline R/show.R]: Should add an explicit test that this works correctly with max_length specified and not.

File: R/expressions.R [hunk 3]
```diff
@@ -59,7 +59,7 @@ math.exp <- function (e1, e2, operator) {
         ## Because of how this function is invoked, get the offending expression
         ## from the call before this one
         halt("Invalid expression (probably a reference to a variable that doesn't exist): ",
-            deparseAndTruncate(tail(sys.calls(), 2)[[1]]))
+             deparseAndFlatten(tail(sys.calls(), 2)[[1]], max_length = NULL))
```
Comment (nealrichardson) [inline R/expressions.R]: One more NULL to remove

File: R/show.R (hunk 1 above)
Comment (jonkeane) [inline R/show.R]: Done.

Comment (malecki) [inline R/show.R]: https://github.com/Crunch-io/rcrunch/issues/46

**Comment by nealrichardson** [comment]
LGTM; just need to add a bullet to NEWS.md

**Pull request merged**